src/components/Navbar/SettingsDrawer/PrayerTimesSection.tsx

```diff
@@ -21,36 +28,76 @@ const calculationMethodOptions = generateSelectOptions(
+          setGeoLocation({
+            latitude: data.coords.latitude,
+            longitude: data.coords.longitude,
+          }),
+        );
+        if (onSuccess) onSuccess();
+      },
+      () => {
+        dispatch(setGeoPermission(GeoPermission.Denied));
+      },
+    );
+  }
+};
+
 const PrayerTimesSection = () => {
   const { t } = useTranslation('common');
   const dispatch = useDispatch();
   const calculationMethod = useSelector(selectCalculationMethod);
   const madhab = useSelector(selectMadhab);
+  const geoPermission = useSelector(selectGeoPermission);
 
   return (
-    <Section>
-      <Section.Title>{t('prayer-times.title')}</Section.Title>
-      <Section.Row>
-        <Section.Label>{t('prayer-times.calculation-method')}</Section.Label>
-        <Select
-          id="prayer-times-calculation-method"
-          name="calculation-method"
-          options={calculationMethodOptions}
-          value={calculationMethod}
-          onChange={(value) => dispatch(setCalculationMethod(value as CalculationMethod))}
-        />
-      </Section.Row>
-      <Section.Row>
-        <Section.Label>{t('prayer-times.madhab')}</Section.Label>
-        <Select
-          id="prayer-times-madhab"
-          name="madhab"
-          options={madhabOptions}
-          value={madhab}
-          onChange={(value) => dispatch(setMadhab(value as Madhab))}
-        />
-      </Section.Row>
-    </Section>
+    <>
+      <ToastContainer />
+      <Section>
+        <Section.Title>{t('prayer-times.title')}</Section.Title>
+        <Section.Row>
+          <Section.Label>{t('prayer-times.calculation-method')}</Section.Label>
+          <Select
+            id="prayer-times-calculation-method"
+            name="calculation-method"
+            options={calculationMethodOptions}
+            value={calculationMethod}
+            onChange={(value) => dispatch(setCalculationMethod(value as CalculationMethod))}
+          />
+        </Section.Row>
+        <Section.Row>
+          <Section.Label>{t('prayer-times.madhab')}</Section.Label>
+          <Select
+            id="prayer-times-madhab"
+            name="madhab"
+            options={madhabOptions}
+            value={madhab}
+            onChange={(value) => dispatch(setMadhab(value as Madhab))}
+          />
+        </Section.Row>
+        <div className={styles.findLocationContainer}>
+          <Button
+            onClick={() =>
+              setAccurateLocation(dispatch, () => {
+                toast(t('prayer-times.location-updated'));
+              })
+            }
+          >
+            {t('prayer-times.find-accurate-location')}
+          </Button>
+          {geoPermission === GeoPermission.Denied && (
+            <div className={styles.accessDeniedContainer}>
+              <h3 className={styles.accessDenied}>{t('prayer-times.access-denied')}</h3>
+              <p>{t('prayer-times.allow-access')}</p>
+            </div>
+          )}
+        </div>
+      </Section>
```

Comment:
Only this part changed in this JSX + adding `<ToastContainer/>`

changes detected because I wrap these JSX inside a fragment `</>`